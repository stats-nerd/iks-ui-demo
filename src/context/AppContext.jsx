import { onAuthStateChanged } from "firebase/auth";
import { useReducer, useContext, createContext, useEffect } from "react";
import { auth } from "../../config";

// Create a context
const AppContext = createContext();

// Initial state for your context
const initialState = {
  patientInfo: null,
  currentHospitalLogo: null,
  currentAuthUser: null,
  loginDocName: null,
  showWindowPopup: false,
  windowPopupLoaderStatus: false,
};

// Define your reducer function to update the state
const appReducer = (state, action) => {
  switch (action.type) {
    case "UPDATE_PATIENT_INFO":
      return { ...state, patientInfo: action.payload };
    case "UPDATE_HOSPITAL_LOGO":
      return { ...state, currentHospitalLogo: action.payload };
    case "UPDATE_CURRENT_USER":
      return { ...state, currentAuthUser: action.payload };
    case "UPDATE_LOGIN_DOC_NAME":
      return { ...state, loginDocName: action.payload };
    case "UPDATE_WINDOW_POPUP_STATUS":
      return { ...state, showWindowPopup: action.payload };
    case "UPDATE_WINDOW_POPUP_LOADER_STATUS":
      return { ...state, windowPopupLoaderStatus: action.payload };
    default:
      return state;
  }
};

// Create a custom hook to access the context
export const useAppContext = () => {
  return useContext(AppContext);
};

// Create a context provider component
export const AppContextProvider = ({ children }) => {
  const [appContextState, appContextDispatch] = useReducer(
    appReducer,
    initialState
  );

  useEffect(() => {
    // Listen for authentication state changes
    const unsubscribe = onAuthStateChanged(auth, (user) => {
      if (user) {
        // User is signed in
        console.log("User is logged in:", user);
        // Perform actions based on the logged-in user (if needed)

        appContextDispatch({ type: "UPDATE_CURRENT_USER", payload: user });
      } else {
        // No user is signed in
        console.log("No user is logged in");
      }
    });

    // Clean up the subscription when the component unmounts
    return () => {
      unsubscribe();
    };
  }, []);

  return (
    <AppContext.Provider
      value={{
        appContextState,
        appContextDispatch,
      }}
    >
      {children}
    </AppContext.Provider>
  );
};
