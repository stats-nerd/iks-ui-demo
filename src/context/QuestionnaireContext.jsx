import { useReducer, useContext, createContext } from "react";

// Create a context
const QuestionnaireContext = createContext();

// Initial state for your context
const initialState = {
  sdohQuestionnaire: [],
  questionnaireType: null,
};

// Define your reducer function to update the state
const questionnaireReducer = (state, action) => {
  switch (action.type) {
    case "UPDATE_SDOH_QUESTIONNAIRE":
      return { ...state, sdohQuestionnaire: action.payload };
    case "UPDATE_QUESTIONNAIRE_TYPE":
      return { ...state, questionnaireType: action.payload };
    default:
      return state;
  }
};

// Create a custom hook to access the context
export const useQuestionnaireContext = () => {
  return useContext(QuestionnaireContext);
};

// Create a context provider component
export const QuestionnaireContextProvider = ({ children }) => {
  const [quesContextState, quesContextDispatch] = useReducer(
    questionnaireReducer,
    initialState
  );
  // need to check this context
  return (
    <QuestionnaireContext.Provider
      value={{
        quesContextState,
        quesContextDispatch,
      }}
    >
      {children}
    </QuestionnaireContext.Provider>
  );
};
