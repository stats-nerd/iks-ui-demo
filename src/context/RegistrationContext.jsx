import { useReducer, useContext, createContext } from "react";
import {
  sideStepperHeading,
  dl_info,
  insurance_info,
  creditCardMonthOptions,
  creditCardOptions,
  creditCardYearOptions,
  creditCard_info,
  genderOptions,
  insuranceProviderOptions,
  stateOptions,
} from "../constants";

// Create a context
const RegistrationContext = createContext();

// Initial state for your context
const initialState = {
  currentStepperSideHeading: sideStepperHeading,
  currentStepperPage: 1,
  frontDL_Picture: null,
  backDL_Picture: null,
  frontIns_Picture: null,
  backIns_Picture: null,
  frontCredit_Picture: null,
  dl_info,
  insurance_info,
  creditCard_info,
  genderOptions,
  stateOptions,
  insuranceProviderOptions,
  creditCardOptions,
  creditCardMonthOptions,
  creditCardYearOptions,
};

// Define your reducer function to update the state
const registrationReducer = (state, action) => {
  switch (action.type) {
    case "UPDATE_SIDE_STEPPER_HEADING":
      return { ...state, currentStepperSideHeading: action.payload };
    case "UPDATE_STEPPER_PAGE":
      return { ...state, currentStepperPage: action.payload };
    case "UPDATE_FRONT_DL":
      return { ...state, frontDL_Picture: action.payload };
    case "UPDATE_BACK_DL":
      return { ...state, backDL_Picture: action.payload };
    case "UPDATE_FRONT_INS":
      return { ...state, frontIns_Picture: action.payload };
    case "UPDATE_BACK_INS":
      return { ...state, backIns_Picture: action.payload };
    case "UPDATE_FRONT_CREDIT":
      return { ...state, frontCredit_Picture: action.payload };
    case "UPDATE_DL_INFO":
      return { ...state, dl_info: action.payload };
    case "UPDATE_INS_INFO":
      return { ...state, insurance_info: action.payload };
    case "UPDATE_CREDIT_CARD_INFO":
      return { ...state, creditCard_info: action.payload };
    case "UPDATE_STATE_OPTIONS":
      return { ...state, genderOptions: action.payload };
    case "UPDATE_GENDER_OPTIONS":
      return { ...state, stateOptions: action.payload };
    case "UPDATE_INS_PROVIDER_OPTIONS":
      return { ...state, insuranceProviderOptions: action.payload };
    case "UPDATE_CREDIT_CARD_OPTIONS":
      return { ...state, creditCardOptions: action.payload };
    case "UPDATE_CREDIT_CARD_MONTH_OPTIONS":
      return { ...state, creditCardMonthOptions: action.payload };
    case "UPDATE_CREDIT_CARD_YEAR_OPTIONS":
      return { ...state, creditCardYearOptions: action.payload };
    default:
      return state;
  }
};

// Create a custom hook to access the context
export const useRegistrationContext = () => {
  return useContext(RegistrationContext);
};

// Create a context provider component
export const RegistrationContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(registrationReducer, initialState);

  return (
    <RegistrationContext.Provider
      value={{
        state,
        dispatch,
      }}
    >
      {children}
    </RegistrationContext.Provider>
  );
};
