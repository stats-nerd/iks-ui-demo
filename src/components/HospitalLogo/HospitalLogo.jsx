import { useAppContext } from "../../context/AppContext";

const HospitalLogo = ({ alt = "hospital logo", className }) => {
  const { appContextState } = useAppContext();

  return (
    <img
      src={`data:image/png;base64,${appContextState.currentHospitalLogo}`}
      alt={alt}
      className={className}
    />
  );
};

export default HospitalLogo;
