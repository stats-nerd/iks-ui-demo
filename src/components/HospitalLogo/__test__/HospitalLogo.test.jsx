import React from "react";
import { render } from "@testing-library/react";
import HospitalLogo from "../HospitalLogo";
import { useAppContext } from "../../../context/AppContext";

// here Mocking the useAppContext hook
jest.mock("../../../context/AppContext", () => ({
  useAppContext: jest.fn(),
}));

describe("HospitalLogo component", () => {
  it("renders correctly with default props", () => {
    useAppContext.mockReturnValue({
      appContextState: { currentHospitalLogo: null },
    });

    const { asFragment } = render(<HospitalLogo />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with provided alt and className props", () => {
    useAppContext.mockReturnValue({
      appContextState: { currentHospitalLogo: "mockLogo" },
    });

    const { asFragment } = render(
      <HospitalLogo alt="testAlt" className="testClass" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with the hospital logo from appContextState", () => {
    useAppContext.mockReturnValue({
      appContextState: { currentHospitalLogo: "mockLogo" },
    });

    const { asFragment } = render(<HospitalLogo alt="testAlt" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with default alt text if alt prop is not provided", () => {
    useAppContext.mockReturnValue({
      appContextState: { currentHospitalLogo: "mockLogo" },
    });

    const { asFragment } = render(<HospitalLogo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
