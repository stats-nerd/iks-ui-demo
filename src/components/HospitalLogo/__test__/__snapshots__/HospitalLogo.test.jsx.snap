// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HospitalLogo component renders correctly with default alt text if alt prop is not provided 1`] = `
<DocumentFragment>
  <img
    alt="hospital logo"
    src="data:image/png;base64,mockLogo"
  />
</DocumentFragment>
`;

exports[`HospitalLogo component renders correctly with default props 1`] = `
<DocumentFragment>
  <img
    alt="hospital logo"
    src="data:image/png;base64,null"
  />
</DocumentFragment>
`;

exports[`HospitalLogo component renders correctly with provided alt and className props 1`] = `
<DocumentFragment>
  <img
    alt="testAlt"
    class="testClass"
    src="data:image/png;base64,mockLogo"
  />
</DocumentFragment>
`;

exports[`HospitalLogo component renders correctly with the hospital logo from appContextState 1`] = `
<DocumentFragment>
  <img
    alt="testAlt"
    src="data:image/png;base64,mockLogo"
  />
</DocumentFragment>
`;
